runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java

```diff
@@ -103,121 +97,118 @@
+		Widget loginScreenTwo = client.getWidget(WidgetInfo.LOGIN_CLICK_TO_PLAY_SCREEN_MESSAGE_OF_THE_DAY);
+
+		if (loginScreenOne != null || loginScreenTwo != null) {
 			return;
 		}
 
 		final String option = Text.removeTags(entry.getOption()).toLowerCase();
 		final String target = Text.removeTags(entry.getTarget()).toLowerCase();
-		final int id = entry.getIdentifier();
-		final int type = entry.getType();
 
-		if (target.contains("pouch") && !target.startsWith("rune"))
-		{
-			if (option.equals("deposit-all") && type == 57 && id == 2)
-			{
-				swap(client, "fill", option, target);
-				swap(client, "cancel", option, "", target);
-			}
-			else  if (option.equals("fill") && id != 9)
-			{
-				swap(client, "empty", option, target);
-			}
-		}
 
-		else if (target.contains("ring of dueling") && option.contains("remove"))
-		{
-			if (target.contains("7") || target.contains("5") || target.contains("3") || target.contains("1"))
-			{
-				swap(client, "castle wars", option, target);
+		Widget widgetBankTitleBar = client.getWidget(WidgetInfo.BANK_TITLE_BAR);
+		if (isEssencePouch(target)) {
+			if (widgetBankTitleBar == null || widgetBankTitleBar.isHidden()) { //swap pouches based on empty/full and in bank
+				swap(client, "Empty", option, target);
+			} else {
+				swap(client, "Fill", option, target); //Due to RuneLite issues the "Deposit" menutext will always show even though it is on fill
 			}
-			else if (wearingBindingNeck())
-			{
+		} else if (target.contains("ring of dueling") && option.contains("withdraw")) { //withdraw-1 ring of dueling
+			swap(client, "withdraw-1", option, target);
+		} else if (target.contains("binding necklace") && option.contains("withdraw")) { //withdraw-1 binding necklace
+			swap(client, "withdraw-1", option, target);
+		}
+		else if (target.contains("ring of dueling") && option.contains("remove")) {
+			if (client.getLocalPlayer().getWorldLocation().getRegionID() != 10315) { //changes duel ring teleport options based on location
 				swap(client, "duel arena", option, target);
+			} else if (client.getLocalPlayer().getWorldLocation().getRegionID() == 10315) {
+				swap(client, "castle wars", option, target);
 			}
 		}
-
-		else if (target.contains("altar") && option.contains("craft")) // Don't accidentally click the altar to craft
+			else if(target.contains("crafting cape") && option.contains("remove")) { //teleport for crafting cape
+				swap(client, "Teleport", option, target);
+			} else if(target.contains("max cape") && option.contains("remove")) { //teleport for max cape
+				swap(client, "Crafting Guild", option, target);
+			}
+		 else if (target.contains("altar") && option.contains("craft")) // Don't accidentally click the altar to craft
 		{
 			hide(option, target, true);
-		}
-		else if (target.contains("pure") && option.contains("use")) // Don't accidentally use pure essence on altar
+		} else if (target.contains("pure")) // Don't accidentally use pure essence on altar
```

Comment:
why change this, you don't have to run the hide code on pure essence in bank/wherever

Reply:
Oops, didn't mean to